api/krusty/sortordertransformer_test.go

```diff
@@ -0,0 +1,215 @@
+metadata:
+  name: pear
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: apple
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: quince
+---
+apiVersion: v1
+kind: Ingress
+metadata:
+  name: durian
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: apricot
+`
+
+func TestCustomOrdering(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+	th.WriteK("base", `
+resources:
+- resources.yaml
+
+sortOptions:
+  order: legacy
+  legacySortOptions:
+    orderFirst:
+    - MutatingWebhookConfiguration
+    - ValidatingWebhookConfiguration
+    - Namespace
+    - ResourceQuota
+    - StorageClass
+    - CustomResourceDefinition
+    - ServiceAccount
+    - PodSecurityPolicy
+    - Role
+    - ClusterRole
+    - RoleBinding
+    - ClusterRoleBinding
+    - ConfigMap
+    - Secret
+    - Endpoints
+    - Service
+    - LimitRange
+    - PriorityClass
+    - PersistentVolume
+    - PersistentVolumeClaim
+    - Deployment
+    - StatefulSet
+    - CronJob
+    - PodDisruptionBudget
+`)
+	th.WriteF("base/resources.yaml", sortOrderResources)
+
+	th.AssertActualEqualsExpected(
+		th.Run("base", th.MakeDefaultOptions()),
+		`
+apiVersion: v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: pomegranate
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: apple
+---
+apiVersion: v1
+kind: Role
+metadata:
+  name: banana
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: apricot
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: quince
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: papaya
+---
+apiVersion: v1
+kind: LimitRange
+metadata:
+  name: peach
+---
+apiVersion: v1
+kind: Deployment
+metadata:
+  name: pear
+---
+apiVersion: v1
+kind: Ingress
+metadata:
+  name: durian
+`)
+}
+
+func TestDefaultLegacyOrdering(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+	th.WriteK("base", `
+resources:
+- resources.yaml
+
+sortOptions:
+  order: legacy
+`)
+	th.WriteF("base/resources.yaml", sortOrderResources)
+	th.AssertActualEqualsExpected(
+		th.Run("base", th.MakeDefaultOptions()),
+		`
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: apple
+---
+apiVersion: v1
+kind: Role
+metadata:
+  name: banana
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: apricot
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: quince
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: papaya
+---
+apiVersion: v1
+kind: LimitRange
+metadata:
+  name: peach
+---
+apiVersion: v1
+kind: Deployment
+metadata:
+  name: pear
+---
+apiVersion: v1
+kind: Ingress
+metadata:
+  name: durian
+---
+apiVersion: v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: pomegranate
+`)
+}
+
+func TestFIFOOrdering(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+	th.WriteK("base", `
+resources:
+- resources.yaml
+
+sortOptions:
+  order: fifo
+`)
+	th.WriteF("base/resources.yaml", sortOrderResources)
+	th.AssertActualEqualsExpected(
+		th.Run("base", th.MakeDefaultOptions()), sortOrderResources)
+}
```

Comment:
Can we add tests for some unhappy cases? e.g. when order is `fifo` but `legacySortOptions` is set